core/src/test/java/apoc/merge/MergeTest.java

```diff
@@ -136,23 +124,59 @@ public void testMergeWithEmptyIdentityPropertiesShouldFail() {
             }
         }
     }
-    
+
+    @Test
+    public void testMergeNodeWithNullLabelsShouldFail() {
+        try {
+            testCall(db, "CALL apoc.merge.node([null, null], {name:'John'}) YIELD node RETURN node",
```

Comment:
It would be good to also have a test where not all labels are null, like `CALL apoc.merge.node(['Person', null], {name:'John'})...`